---
title: "Research"
author: ""
output: html_document
---

```{r, echo= FALSE, results = "asis"}
cat("<br>")
```

```{r, echo= FALSE, message= FALSE, warning=FALSE}
#++++++++++++++++++++++++++++++++++
# rquery.wordcloud() : Word cloud generator
# - http://www.sthda.com
#+++++++++++++++++++++++++++++++++++
# x : character string (plain text, web url, txt file path)
# type : specify whether x is a plain text, a web page url or a file path
# lang : the language of the text
# excludeWords : a vector of words to exclude from the text
# textStemming : reduces words to their root form
# colorPalette : the name of color palette taken from RColorBrewer package, 
# or a color name, or a color code
# min.freq : words with frequency below min.freq will not be plotted
# max.words : Maximum number of words to be plotted. least frequent terms dropped
# value returned by the function : a list(tdm, freqTable)
rquery.wordcloud <- function(x, type=c("text", "url", "file"), 
                             lang="english", excludeWords=NULL, 
                             textStemming=FALSE,  colorPalette="Dark2",
                             min.freq=3, max.words=200)
{ 
  library("tm")
  library("SnowballC")
  library("wordcloud")
  library("RColorBrewer") 
  
  if(type[1]=="file") text <- readLines(x)
  else if(type[1]=="url") text <- html_to_text(x)
  else if(type[1]=="text") text <- x
  
  # Load the text as a corpus
  docs <- Corpus(VectorSource(text))
  # Convert the text to lower case
  docs <- tm_map(docs, content_transformer(tolower))
  # Remove numbers
  docs <- tm_map(docs, removeNumbers)
  # Remove stopwords for the language 
  docs <- tm_map(docs, removeWords, stopwords(lang))
  # Remove punctuations
  docs <- tm_map(docs, removePunctuation)
  # Eliminate extra white spaces
  docs <- tm_map(docs, stripWhitespace)
  # Remove your own stopwords
  if(!is.null(excludeWords)) 
    docs <- tm_map(docs, removeWords, excludeWords) 
  # Text stemming
  if(textStemming) docs <- tm_map(docs, stemDocument)
  # Create term-document matrix
  tdm <- TermDocumentMatrix(docs)
  m <- as.matrix(tdm)
  v <- sort(rowSums(m),decreasing=TRUE)
  d <- data.frame(word = names(v),freq=v)
  # check the color palette name 
  if(!colorPalette %in% rownames(brewer.pal.info)) colors = colorPalette
  else colors = brewer.pal(8, colorPalette) 
  # Plot the word cloud
  set.seed(1234)
  wordcloud(d$word,d$freq, min.freq=min.freq, max.words=max.words,
            random.order=FALSE, rot.per=0.35, 
            use.r.layout=FALSE, colors=colors)
  
  invisible(list(tdm=tdm, freqTable = d))
}
#++++++++++++++++++++++
# Helper function
#++++++++++++++++++++++
# Download and parse webpage
html_to_text<-function(url){
  library(RCurl)
  library(XML)
  # download html
  html.doc <- getURL(url)  
  #convert to plain text
  doc = htmlParse(html.doc, asText=TRUE)
  # "//text()" returns all text outside of HTML tags.
  # We also don’t want text such as style and script codes
  text <- xpathSApply(doc, "//text()[not(ancestor::script)][not(ancestor::style)][not(ancestor::noscript)][not(ancestor::form)][not(ancestor::strong)]", xmlValue)
  # Format text vector into one character string
  return(paste(text, collapse = " "))
}




########################################

url <- "https://robinsonlabuzh.github.io/publications.html"

exWord <- c("doi", "robinson", "pmid", "science", "hughes", "statham",
            "bioinformatics", "krogan", "bmc", "publications", "zhang", 
            "sci", "strbenac", "data", "berge", "synthetic", "journal",
            "müller", "liu", "wong", "")

rquery.wordcloud(x=url, type="url", excludeWords = exWord, 
                 min.freq = 2, lang = "english")


```

#### **Overall philosophy of interdisciplinary methods research**

High-throughput sequencing technologies are revolutionizing the way genomes, transcriptomes and epigenomes are studied. Generating, processing and interpreting this unprecedented magnitude of complex data and integrating multiple dimensions requires considerable research effort, for biologists, statisticians and computer scientists alike. Our group develops statistical methods for interpreting high-throughput sequencing and other genomics technologies in areas such as genome sequencing, gene expression and regulation and analysis of epigenomes. Research projects encompass three interlaced themes:

1. contribute to relevant genome-scale experimental design and data analysis challenges in close collaboration with biologists;

2. develop statistical methods and software tools for the interpretation and integration of multiple types of genome-scale data;

3. perform technical investigations to understand the performance of the technologies used to generate and of the algorithms used to process the data.


```{r, echo= FALSE, results = "asis"}
cat("<br>")
```

<!-- ```{r, echo= FALSE, results = "asis"} -->
<!-- # This is to add two blank lines -->
<!-- cat("<br><br>") -->
<!-- ``` -->

<!-- --- -->
<!-- #### **Research area 1** [ [**_Publications on this area_**](file://localhost/Users/ruizhu/Documents/work/robinsonlabuzh.github.io/publications.html)] [ [**_Tools on this area_**](file://localhost/Users/ruizhu/Documents/work/robinsonlabuzh.github.io/tools.html)] -->

<!-- ```{r echo = FALSE} -->
<!-- plot(rpois(100, 50), rpois(100, 25),  -->
<!--      cex = runif(100, min = 0, max = 5),  -->
<!--      col = rgb(0.2, runif(100), 1, 0.5), -->
<!--      xlab = "", ylab = "", pch = 19, -->
<!--      axes = FALSE) -->
<!-- ``` -->

<!-- Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project.   -->

<!-- ```{r, echo= FALSE, results = "asis"} -->
<!-- # add one blank line -->
<!-- cat("<br>") -->
<!-- ``` -->

#### **Our formula for embedded methodological research**

Our team of computational scientists is fully immersed in and motivated by biological investigations.  Many of us are statisticians, thus it is critical to be involved at the experimental design of these studies. Overall, our focus is on methodology development, but a large amount of our research time is ultimately spent with our collaborators investigating questions, either technical or biological, in a specific dataset. We do not necessarily drive the biological questions, but our expertise drives the method development that is fundamental to answering them.

```{r, echo= FALSE, results = "asis"}
cat("<br>")
```

#### **Open science and reproducibility**


Our group maintains a firm stance on open and reproducible science. The default policy is that we: i) produce documented open source software for our developed statistical and computational frameworks (typically through the Bioconductor project); ii) create code repositories for manuscript analyses; and, iii) post manuscripts as preprints. In many subfields, modern biologists are transitioning to faster, more transparent modes of scientific dissemination and we fully support these initiatives (e.g., ASAPBio).



```{r, echo= FALSE, results = "asis"}
# add one blank line
cat("<br>")
```

<!-- #### **Research area 3** [ [**_Publications on this area_**](file://localhost/Users/ruizhu/Documents/work/robinsonlabuzh.github.io/publications.html)] [ [**_Tools on this area_**](file://localhost/Users/ruizhu/Documents/work/robinsonlabuzh.github.io/tools.html)] -->

<!-- ```{r, echo= FALSE, results = "asis"} -->
<!-- # create a two column style -->
<!-- l1 <- paste0("<style> \n") -->
<!-- l2 <- paste0("\t", ".col2 {\n") -->
<!-- l3 <- paste0("\t", "\t","columns: 2 200px; \n") -->
<!-- l4 <- paste0("\t", "\t","-webkit-columns: 2 200px; \n") -->
<!-- l5 <- paste0("\t", "\t","-moz-columns: 2 200px;") -->
<!-- l6 <- paste0("\t", "} \n") -->
<!-- l7 <- paste0("</style> \n") -->
<!-- style <- do.call(paste, mget(paste0("l",1:7))) -->
<!-- cat(style) -->
<!-- ``` -->


<!-- ```{r, echo= FALSE, results = "asis"} -->
<!-- # start a division -->
<!-- cat("<div class='col2'>") -->
<!-- ``` -->


<!-- ```{r, echo=FALSE} -->
<!-- # one figure in the left column -->
<!-- plot(rpois(100, 50), rpois(100, 25),  -->
<!--      cex = runif(100, min = 0, max = 5),  -->
<!--      col = rgb(0.2, runif(100), 1, 0.5), -->
<!--      xlab = "", ylab = "", pch = 19, -->
<!--      axes = FALSE) -->
<!-- # text in the right column -->
<!-- ``` -->

<!-- ```{r, echo= FALSE, results = "asis"} -->
<!-- cat("<br>") -->
<!-- ``` -->

<!-- Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project. Description of the project.  -->

```{r, echo= FALSE, results = "asis"}
# end of a division (CSS code)
cat("</div>")
```